### DISCUSSION TITLE
render login form in login selector

### DISCUSSION THREAD

**Discussion by do9xe** [indico/flask-multipass#79] [General]
Currently I'm playing around with Flask-Multipass for a customers project and I wanted to implement the login selector page in a way that it is possible to either directly login with credentials via a login form or alternatively via SSO and other providers. The usecase would be a WebApp where you either login with username/password or SSO-providers. This is a random example from the internet:

![](https://assets-global.website-files.com/621f84dc15b5ed16dc85a18a/6234a024fad509aa93b428dd_5f2c72b88e63c617c8e4154e_ezgif.com-gif-maker.gif)

I played around and was able to make it work by creating a form object in the `Multipass._login_selector()` method and passing it to the template rendering. I could imagine to habe a config option like `MULTIPASS_LOGIN_PRIMARY_PROVIDER` and then the form is created from this providers form class, if available.

If you like the idea I'd craft something and provide it as a PR to the code, to my knowledge it shouldn't be that much code.

**Comment by ThiefMaster** [comment]
You mean something like what we have on this page? https://agenda.linearcollider.org/login/

If yes, you can check the code here:

- https://github.com/indico/indico/blob/master/indico/modules/auth/templates/login_page.html
- https://github.com/indico/indico/blob/master/indico/modules/auth/controllers.py#L113

**Reply by do9xe** [reply]
Basically, yes. I managed to build this by overwriting the Multipass class and adding a form attribute to the `_login_selector` method, which works fine, I just thought it might be interesting to support this natively. If this is out of scope for the project I'm fine with that.
